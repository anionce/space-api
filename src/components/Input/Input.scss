@import '../../assets/styles/variables';

.input-container {
	display: flex;
	flex-direction: column;
	align-items: center;
}
