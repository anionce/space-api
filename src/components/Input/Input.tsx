import './Input.scss';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import React from 'react';

type inputProps = {
	onInputChange?: (event: React.FormEvent<HTMLInputElement>) => void;
	onTitleChanged?: (event: React.FormEvent<HTMLInputElement>) => void;
	searchText?: string | undefined;
	serial?: string | undefined;
};

export const Input = ({ onInputChange, searchText, onTitleChanged, serial }: inputProps) => {
	const { t } = useTranslation();
	const { pathname } = useLocation();

	const isForm: Boolean = pathname.includes('add') || pathname.includes('update');

	return (
		<div className='input-container'>
			<label htmlFor='input'>{isForm ? t('space_api.input.title') : t('space_api.input.search')}</label>
			<input
				id='input'
				maxLength={100}
				onChange={isForm ? onTitleChanged : onInputChange}
				value={isForm ? serial : searchText}></input>
		</div>
	);
};
