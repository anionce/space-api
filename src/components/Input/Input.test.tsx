import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { Input } from './Input';

test('renders correcly', () => {
	const { container } = customRender(<Input />);

	expect(container).toMatchSnapshot();
});

test('renders elements correctly', () => {
	customRender(<Input />);

	expect(screen.getByLabelText(/Search/i)).toBeInTheDocument();
});
