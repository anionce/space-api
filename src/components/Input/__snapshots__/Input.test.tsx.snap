// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="input-container"
  >
    <label
      for="input"
    >
      Search
    </label>
    <input
      id="input"
      maxlength="100"
      value=""
    />
  </div>
</div>
`;
